docs/standard/commandline/model-binding.md

```diff
@@ -0,0 +1,225 @@
+---
+title: How to bind arguments to handlers in System.CommandLine
+description: "Learn how to do model-binding in apps that are built with the System.Commandline library."
+ms.date: 02/03/2022
+no-loc: [System.CommandLine]
+helpviewer_keywords:
+  - "command line interface"
+  - "command line"
+  - "System.CommandLine"
+ms.topic: how-to
+---
+# How to bind arguments to handlers in System.CommandLine
+
+[!INCLUDE [scl-preview](../../../includes/scl-preview.md)]
+
+The process of parsing arguments and providing them to command handler code is called *model binding*. `System.CommandLine` has the ability to bind many argument types built in. For example, integers, enums, and file system objects such as `FileInfo` and `DirectoryInfo` can be bound. `FileInfo` and `DirectoryInfo` are examples of a general convention whereby any type that has a constructor that takes a single `string` parameter can be bound without your having to write any custom code. Several `System.CommandLine` types can also be bound.
+
+## Built-in argument validation
+
+Arguments have expected types and [arity](syntax.md#argument-arity). `System.CommandLine` rejects arguments that don't match these expectations.
```

Comment:
Do we need to define arity?

Reply:
I think the following example explains it but I can add more text if you think it needs it.